In [68]:
import matplotlib.pyplot as plt
import pandas as pd
df = pd.read_csv("Y:/BTS/BTS_DataScienceFoundations/Session_2/2_titanic_dataset.csv")
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

1. How many people are on the ship?

In [6]:
df.shape

(891, 12)

There are 891 people on the ship.

2. How many survived?

In [7]:
sum(df["Survived"]==1)

342

342 people survived

3. What is the survive ratio?

In [11]:
sum(df["Survived"]==1)/len(df["PassengerId"])

0.38383838383838381

38% of all passengers survived

4. What is the ratio between man and woman?

In [22]:
sum(df["Sex"] == "male")/len(df["PassengerId"])

0.6475869809203143

65% are man, 35% woman

5. What is the survive ratio of man?

In [39]:
counter = 0

for index,row in df.iterrows():
    if row["Sex"]=="male" and row["Survived"]==1:
        counter += 1

counter/sum(df["Sex"] == "male")

0.18890814558058924

19% of man survived

6. What is the survive ratio of woman?

In [41]:
counter = 0

for index,row in df.iterrows():
    if row["Sex"]=="female" and row["Survived"]==1:
        counter += 1

counter/sum(df["Sex"] == "female")

0.7420382165605095

74% of woman survived

7. How many people were below 18 years old did not survive?

In [47]:
sum(df["Age"] <18)

113

113 of kids on the ship

8. How many people were below 18 years old did not survive?

In [43]:
counter = 0

for index,row in df.iterrows():
    if row["Age"]<18 and row["Survived"]==1:
        counter += 1

counter/sum(df["Age"] <18)

0.53982300884955747

only 54% of kids survived?!?!

9. What is the oldest and youngest age of the person survived?

In [49]:
df["Age"].min()

0.41999999999999998

In [45]:
df["Age"].max()

80.0

Youngest is 5 months old, oldest is 80 years old

10. How many of them are single household ratio?

In [51]:
counter = 0

for index,row in df.iterrows():
    if row["SibSp"]==0 and row["Parch"]==0:
        counter += 1
        
counter/len(df["PassengerId"])

0.6026936026936027

60% are single household

11. What is the avg number sibling/spouse on the ship?

In [52]:
df["SibSp"].mean()

0.5230078563411896

Passengers have on average half of a sibling or spouse.

12. What is the avg no of parent/childen on the ship?

In [53]:
df["Parch"].mean()

0.38159371492704824

Passengers have on average 0.4 parent or kid.

13. How many people are in Class 1/2/3?

In [54]:
sum(df["Pclass"]==1)

216

In [55]:
sum(df["Pclass"]==2)

184

In [56]:
sum(df["Pclass"]==3)

491

Class 1 has 216 people, Class 2 has 184 people, Class 3 has 491 people

14. Does anybody has no ticket?

In [60]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Everyone had a ticket

15. Is Rose on the ship?

In [67]:
sum(df["Name"]=="Rose")

0

No Rose on the ship.

In [65]:
sum(df["Name"]=="Jack")

0

No Jack on the ship.

16. What is the average fare?

In [66]:
df["Fare"].mean()

32.2042079685746

17. What the fare distrubution?

In [73]:
import matplotlib.pyplot as plt
plt.plot(df["Fare"])
plt.show

<function matplotlib.pyplot.show>